scim/scim.bal

```diff
@@ -0,0 +1,576 @@
+    string nickName?;
+    string profileUrl?;
+    Email[] emails?;
+    Address[] addresses?;
+    Phone[] phoneNumbers?;
+    Ims[] ims?;
+    Role[] roles?;
+    Photo[] photos?;
+    string userType?;
+    string title?;
+    string preferredLanguage?; 
+    string locale?;
+    string timezone?; 
+    boolean active?;
+    Group[] groups?;
+    Certificate[] x509Certificates?;
+    Meta meta?;
+    SCIMEnterpriseUser urn\:ietf\:params\:scim\:schemas\:extension\:enterprise\:2\.0\:User?;
+    
+};
+
+public type PatchOp record {
+    string op;
+    Value value;
+    
+};
+
+public type UserPatch record {
+    string[] schemas;
+    PatchOp[] Operations;
+    
+};
+
+public type UserSearch record {
+    string[] schemas;
+    string[] attributes;
+    string filter;
+    string domain?;
+    int startIndex?;
+    int count?;
+    
+};
+
+public type GroupResponse record {
+    int totalResults?;
+    int startIndex?;
+    int itemsPerPage?;
+    string[] schemas?;
+    GroupResource[] Resources?;
+    
+};
+
+public type GroupResource record {
+    string displayName?;
+    Meta meta;
+    Member[] members?;
+    Role[] roles?;
+    string id?;
+    
+};
+
+public type GroupUpdate record {
+    string displayName; 
+    Member[] members?;
+    Meta meta?;
+    Role roles?;
+
+};
+
+public type GroupPatch record {
+    string[] schemas;
+    PatchOpGroup[] Operations;
+    
+};
+
+public type PatchOpGroup record {
+    string op;
+    ValueGroup value;
+    
+};
+
+public type ValueGroup record {
+    Member[] members?;
+    Meta meta?;
+    Role roles?;
+    
+};
+
+public type GroupSearch record {
+    string[] schemas;
+    string filter;
+    int startIndex?;
+
+};
+
+public type Operation record {
+    string method;
+    string path;
+    string bulkId?;
+    SCIMUser|BulkUserUpdate|BulkUserReplace|SCIMGroup|BulkGroupUpdate|BulkGroupReplace data?;
+    
+};
+
+public type Bulk record {
+    int failOnErrors?;
+    string[] schemas;
+    Operation[] Operations;
+    
+};
+
+public type BulkUserReplace record {
+    string[] schemas;
+    string username;
+    Name name;
+    
+};
+
+public type BulkUserUpdate record {
+    string op;
+    string path;
+    string value;
+    
+};
+
+public type BulkGroupReplace record {
+    string displayName;
+    MemberReplace[] members;
+    
+};
+
+public type MemberReplace record {
+    string value;
+    string display;
+    
+};
+
+public type BulkGroupUpdate record {
+    string op;
+    string path;
+    json[] value;
+    
+};
+
+public type OperationResponse record {
+    string method;
+    string location?;
+    string bulkId;
+    Status status;
+    string detail?;
+    OpResponse|string response?;
+    
+};
+
+public type Status record {
+    int code;
+    
+};
+
+public type OpResponse record {
+    string[] schemas; 
+    
+};
+
+public type BulkResponse record {
+    string[] schemas;
+    OperationResponse[] Operations;  
+    
+};
+
+
+public class Client{
```

Comment:
Add code comments. Check how the code comments are added in other ballerina modules and let's follow the same approach